### PR TITLE
EXPERIMENTAL_Table motion

### PR DESCRIPTION
**Pull request by matheusps**
#### 🔥Added table motions

♻️Extensible and reusable.
🚫No external libraries. 
🖥Respects the user accessibility motion definitions.

#### What problem is this solving?

[Running workspace](https://tablemotion--cosmetics1.myvtex.com/admin/app/collections)

#### How should this be manually tested?

Clone the repo and `yarn && yarn start`. You should test for reduced and not reduced motion.

#### Screenshots or example usage

On density change
density

On bulk actions toggle
bulk

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EXPERIMENTAL_TableTree/Tree/Node.tsx [hunk 1]
```diff
@@ -8,6 +8,17 @@ import CollapseToggle from './CollapseToggle'
 import { Checkboxes } from '../../EXPERIMENTAL_useCheckboxTree/types'
 import { Column } from '../../EXPERIMENTAL_Table'
 import { Density } from '../../EXPERIMENTAL_Table/hooks/useTableMeasures'
+import useTableMotion from '../../EXPERIMENTAL_Table/hooks/useTableMotion'
+
+const TRANSITIONS = [
```
Comment (kevinch) [inline react/components/EXPERIMENTAL_TableTree/Tree/Node.tsx]: Are those the same as `BulkActions/index.tsx`?

Comment (matheusps) [reply react/components/EXPERIMENTAL_TableTree/Tree/Node.tsx]: It was the same as rows, nice catch!

File: react/components/EXPERIMENTAL_Table/hooks/useTableMotion.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    mql.addEventListener('change', throttle(handleChange, 200))
+    return () => {
+      mql.removeEventListener('change', handleChange)
+    }
+  }, [])
+  return reduced
+}
+
+type transition = {
```
Comment (rerissondaniel) [inline react/components/EXPERIMENTAL_Table/hooks/useTableMotion.ts]: I always like to use uppercase for types, but I know that is not enforced

Comment (matheusps) [reply react/components/EXPERIMENTAL_Table/hooks/useTableMotion.ts]: I agree with you in this case. Nice catch, btw. When my type expresses a callback or function I use camelCase though

**Pull request merged**